svm/src/transaction_processor.rs

```diff
@@ -601,12 +593,7 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
         let fee_details = if lamports_per_signature == 0 {
```

Comment:
can we remove this `if` now?

Reply:
I'll try. I suspect the 0 `lamports_per_signature` is being used in tests, and it'll need bunch of rework there.

Reply:
cc @tao-stones for awareness.

We have both run into how difficult it is to entirely remove the 0 fee stuff. Even if we cannot remove now, we should aim to remove it from SVM's responsibility 😄

Reply:
I've removed it, and the tests passed locally. Let's see if CI is happy with it.

Reply:
Removing this is causing `partitions` tests to fail. I can try to move it out of SVM, so that it's contained in monorepo.